### ISSUE TITLE
OverflowError in QTreeView while creating EditorSession

### ISSUE THREAD

**Issue by sagikazarmark** [mcedit/mcedit2#53]
I downloaded the latest stable release and built it on Ubuntu 14.10.

The application starts fine, founds the maps correctly (and even draws the overview).

When I click the Edit button it exits with the following backtrace:

```
Logging level set
Logging to /tmp/mcedit2-2.0.0alpha/MCEdit User Data/mcedit.log
Setting up console handler
[WARNING][editorapp.py:77]:UserFilesDirectory: /tmp/mcedit2-2.0.0alpha/MCEdit User Data
Exception AttributeError: "'RevisionHistory' object has no attribute 'nodes'" in  ignored
[ERROR][main.py:125]:Unhandled Exception: 
	
Traceback (most recent call last):
  File "/tmp/mcedit2-2.0.0alpha/src/mcedit2/editorapp.py", line 567, in editWorldFromList (self is a MCEditApp)
    self.loadFile(filename)
  File "/tmp/mcedit2-2.0.0alpha/src/mcedit2/editorapp.py", line 534, in loadFile (self is a MCEditApp)
    session = EditorSession(filename, self.worldList.getSelectedIVP(), readonly=readonly)
  File "/tmp/mcedit2-2.0.0alpha/src/mcedit2/editorsession.py", line 169, in __init__ (self is a EditorSession)
    self.playerPanel = PlayerPanel(self)
  File "/tmp/mcedit2-2.0.0alpha/src/mcedit2/panels/player.py", line 61, in __init__ (self is a PlayerPanel)
    self.setSelectedPlayerIndex(0)
  File "/tmp/mcedit2-2.0.0alpha/src/mcedit2/panels/player.py", line 147, in setSelectedPlayerIndex (self is a PlayerPanel)
    self.setSelectedPlayerUUID(UUID)
  File "/tmp/mcedit2-2.0.0alpha/src/mcedit2/panels/player.py", line 156, in setSelectedPlayerUUID (self is a PlayerPanel)
    self.updateNBTTree()
  File "/tmp/mcedit2-2.0.0alpha/src/mcedit2/panels/player.py", line 113, in updateNBTTree (self is a PlayerPanel)
    self.nbtEditor.setRootTag(self.selectedPlayer.rootTag)
  File "/tmp/mcedit2-2.0.0alpha/src/mcedit2/widgets/nbttree/nbteditor.py", line 108, in setRootTag (self is a NBTEditorWidget)
    self.treeView.expandToDepth(0)
OverflowError
```

**Comment by sagikazarmark** [comment]
Ah, sorry, I just realised it is only alpha.

**Comment by codewarrior0** [comment]
It might be related to this issue, which is 64-bit only.

https://bugreports.qt.io/browse/PYSIDE-103

**Comment by sagikazarmark** [comment]
Ah, I am going to try to build it with 32-bit libraries. Thanks

**Comment by sagikazarmark** [comment]
Is there a chance this one can be solved from within mcedit? I can't really manage to build 32-bit libraries.

**Comment by codewarrior0** [comment]
I have no idea. I can't reproduce the problem. It will be up to Linux users to figure out what's going on.

**Comment by naor2013** [comment]
Can you please fix it??
It happens on Mac too and Mac and Linux users just can't open worlds on MCEdit

**Comment by codewarrior0** [comment]
Again, I have no idea.

If you can confirm that this happens on a 32-bit Python, then we can rule out PYSIDE-103 as the cause.

**Comment by sagikazarmark** [comment]
I haven't tried building it on 32bit (yet).

**Comment by ishitatsuyuki** [comment]
https://github.com/PySide/shiboken2/pull/1
This is the possible fix. (requires repackage)

**Comment by ishitatsuyuki** [comment]
Applied patch. No result. Possibly your fault.

**Comment by codewarrior0** [comment]
32-bit python?

**Comment by sagikazarmark** [comment]
The PR mentions 64bit.

**Comment by codewarrior0** [comment]
Once again, if you can confirm that this OverflowError happens on a 32-bit Python, then we can rule out PYSIDE-103 as the cause.

**Comment by ishitatsuyuki** [comment]
Couldn't reproduce on 32bit but there is a more serious error after that(segfault).
Also I need to mention that PYSIDE-103 affects windows, doesn't it?

**Comment by codewarrior0** [comment]
You're right, it should only affect windows since `long` is 32-bits when compiling with 64-bit MSVC. And that's not even a problem since I use mingw builds of PySide. 

I've got no idea where the OverflowError is coming from. At least, I learned that PySide will raise OverflowError when passing a Python int/long to a function whose integer data type is to short to represent it, and likewise with Python float and C float. Maybe it's something in your level files that I can reproduce. Would you mind posting your level.dat?

**Comment by ishitatsuyuki** [comment]
https://drive.google.com/file/d/0B8jwchcNlNEHZUlKcld0eDlHcm8/view?usp=sharing
Here you are.
(repeating) Env: Arch Linux x64 Sucking intel, Forge 1.7 70 mods

**Comment by codewarrior0** [comment]
Oh whoops, it's a server level, so there's no player in the level.dat. You probably don't want to upload the whole thing then, but how about the `playerdata` folder?

**Comment by ishitatsuyuki** [comment]
Uploaded. It's singleplayer. Maybe due to the system change in MC1.3.x?

**Comment by codewarrior0** [comment]
Can't reproduce. Pretty sure it's not the level, then.

![image](https://cloud.githubusercontent.com/assets/307683/8897607/e3ac646a-33b3-11e5-9ea3-ba485753644b.png)

Oh well. I'll post again when I get around to working on the NBTEditor some more.

**Comment by zatherz** [comment]
Bumping. Exact same issue, 64 bit Arch Linux.

**Comment by Barteks2x** [comment]
I have the same issue, debian 8 (jessie) 64-bit.

**Comment by codewarrior0** [comment]
Should be fixed in 73447bee690abee2b00ba691e10f99c769b2abec on the latest master.
